src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -119,6 +124,31 @@ ss::future<> service::create_internal_topic() {
     // TODO(Ben): Validate the _schemas topic
 }
 
+ss::future<> service::fetch_internal_topic() {
+    auto offset_res = co_await _client.local().list_offsets(
+      model::schema_registry_internal_tp);
+    const auto& topics = offset_res.data.topics;
+    if (topics.size() != 1 || topics.front().partitions.size() != 1) {
+        auto ec = kafka::error_code::unknown_topic_or_partition;
+        throw kafka::exception(ec, make_error_code(ec).message());
+    }
+    const auto& partition = topics.front().partitions.front();
+    if (partition.error_code != kafka::error_code::none) {
+        auto ec = partition.error_code;
+        throw kafka::exception(ec, make_error_code(ec).message());
+    }
+
+    auto max_offset = partition.offset;
+    vlog(plog.debug, "Schema registry: _schemas max_offset: {}", max_offset);
+
+    co_await make_client_fetch_batch_reader(
+      _client.local(),
+      model::schema_registry_internal_tp,
+      model::offset{0},
+      max_offset)
+      .consume(consume_to_store{_store}, model::no_timeout);
```

Comment:
this is cool

Reply:
> this is cool

Thanks. I'm playing around with using `record_batch_reader` for a bunch of things. I see some patterns emerging such as @graphcareful `zip_consumer` (by another name), `client_fetch_reader` is a kind of `join_producer`. There are a couple of places where it's almost used for the JSON serde, too, which I hope to refactor once I get a better feel for those patterns. 

It's a nice interface, if applied in abundance it might allow the composition of components to be much higher up the stack where composition and coordination belong. I'm experimenting.